Gem/Code/Source/Components/NetworkMatchComponent.cpp

```diff
@@ -438,7 +440,9 @@ namespace MultiplayerSample
         {
             if (PlayerIdentityComponent* identity = entityHandle.GetEntity()->FindComponent<PlayerIdentityComponent>())
             {
-                identity->AssignPlayerName(PlayerNameString::format("Player %d", m_nextPlayerId));
+                const PlayerNameString prefixName = AutoAssignedPlayerNamePrefix[m_randomNameGenerator.GetRandom() % AutoAssignedPlayerNamePrefix.size()];
+                const PlayerNameString postfixName = AutoAssignedPlayerNamePostfix[m_randomNameGenerator.GetRandom() % AutoAssignedPlayerNamePostfix.size()];
+                identity->RPC_AssignPlayerName(prefixName+postfixName);
```

Comment:
nit: I would suggest adding at end rand int to lessen chances of name collisions. In the real world, backend identify systems would prevent this. With a small pool of names as in this example, we should be over cautious otherwise its going to be confusing in game when it does happen.